src/VisualStudio/Core/Def/Implementation/ProjectSystem/LinkedFileUtilities.cs

```diff
@@ -152,5 +152,23 @@ private string GetActiveIntellisenseProjectContextInternal(IVsHierarchy hierarch
                 ? intellisenseProjectName as string
                 : null;
         }
+
+        public static bool TryGetItemIdInSharedHierarchy(IVsHierarchy hierarchy, uint itemId, IVsHierarchy sharedHierarchy, out uint itemIdInSharedHierarchy)
```

Comment:
Discussed offline: let's make this return the sharedHierarchy instead of accept it redundantly.